codex/contracts/deployment.nim

```diff
@@ -12,8 +12,8 @@ const defaultFile = "vendor" / "codex-contracts-eth" / "deployment-localhost.jso
 ## been exported with Hardhat deploy.
 ## See also:
 ## https://github.com/wighawag/hardhat-deploy/tree/master#6-hardhat-export
-proc deployment*(file = defaultFile): Deployment =
-  Deployment(json: parseFile(file))
+proc deployment*(file: ?string = some defaultFile): Deployment =
```

Comment:
This looks a bit weird to me: an optional parameter with a default value.
Perhaps we can avoid this change; see comments below.

Reply:
Updated the `HostInteractions`, `ClientInteractions`, and `Deployment` constructors.